### PR TITLE
CAT-1937 Ask Brick

### PR DESCRIPTION
**Pull request by p4p4**
**reopens #1810**

- ask a user defined question
- answer is entered in a dialogue
- program is paused while user enters answer
- program is resumed as soon as an answer is provided by the user
- this contains an ActionTest

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/stage/StageActivity.java [hunk 1]
```diff
@@ -118,6 +133,55 @@ public void onCreate(Bundle savedInstanceState) {
...
+	}
+
+	private void showDialog(String question, final AskAction askAction) {
+		pause();
+
+		AlertDialog.Builder alertBuilder = new AlertDialog.Builder(new ContextThemeWrapper(this, android.R.style.Theme_Holo_Dialog));
+		final EditText edittext = new EditText(getContext());
+		alertBuilder.setView(edittext);
+		alertBuilder.setMessage(getContext().getString(R.string.brick_ask_dialog_hint));
+		alertBuilder.setTitle(question + "?");
```
Comment (sjaindl) [inline catroid/src/main/java/org/catrobat/catroid/stage/StageActivity.java]: IMO the ? should not be concatenated here, but entered by the user.
Reason: Right to left languages, or even e.g. Spanish where a reversed question marks are in the front: "¿ question ?"

File: catroid/src/main/res/layout/brick_ask.xml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            android:id="@+id/brick_ask_spinner"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:clickable="false"
+            android:focusable="false"
+            app:layout_newLine="true" />
+
+    </org.catrobat.catroid.ui.BrickLayout>
+
+</LinearLayout>
```
Comment (sjaindl) [inline catroid/src/main/res/layout/brick_ask.xml]: add newline

File: catroid/src/main/java/org/catrobat/catroid/stage/StageActivity.java (hunk 1 above)
Comment (p4p4) [inline catroid/src/main/java/org/catrobat/catroid/stage/StageActivity.java]: you are right! got the same feedback from UX

**Comment by p4p4** [comment]
tests after rebase: [click me](https://jenkins.catrob.at/job/CatroidSinglePackageEmulatorTest/4480/)

everything except
org.catrobat.catroid.test.devices.mindstorms.nxt.SensorTests.testI2CUltrasonicSensor
works

**Pull request merged**